src/interactions/slashCommands/SlashCommandBuilder.ts

```diff
@@ -0,0 +1,122 @@
+import type { APIApplicationCommandOption } from 'discord-api-types/v8';
+import { mix } from 'ts-mixer';
+import { validateMaxOptionsLength, validateRequiredParameters } from './Assertions';
+import { SharedNameAndDescription, SharedSlashCommandOptions } from './SlashCommandOptions';
+import { SlashCommandSubCommandBuilder, SlashCommandSubCommandGroupBuilder } from './SlashCommandSubCommands';
+
+@mix(SharedSlashCommandOptions, SharedNameAndDescription)
+export class SlashCommandBuilder {
+	protected name: string = undefined!;
+
+	protected description: string = undefined!;
+	protected options: ToAPIApplicationCommandOptions[] = [];
+
+	/**
+	 * Returns the final data that should be sent to Discord. You won't need this unless you're manually
+	 * creating slash commands via this builder.
+	 *
+	 * **Note:** Calling this getter will validate required properties based on their conditions.
+	 */
+	public toJSON() {
+		validateRequiredParameters(this.name, this.description, this.options);
+		return {
+			name: this.name,
+			description: this.description,
+			options: this.options.map((option) => option.toJSON()),
+		};
+	}
+
+	// #region Builder methods
```

Comment:
We don't use regions anywhere in other repositories, should we even have them?

Reply:
I'm not a huge fan of regions, usually they indicate that the file is too big and should be split up.

Reply:
I don't even remember why I used regions in this file (I know why I used them in another file, tho it's not needed)